crates/ruff_cli/src/lib.rs

```diff
@@ -43,6 +45,19 @@ impl From<ExitStatus> for ExitCode {
     }
 }
 
+pub fn has_file_changed(paths: &[PathBuf]) -> bool {
+    for path in paths {
+        if let (Some(file_name), Some(suffix)) = (path.file_name(), path.extension()) {
+            if let (Some(file_name), Some(suffix)) = (file_name.to_str(), suffix.to_str()) {
+                if WATCHED_FILES.contains(&file_name) || WATCHED_FILES.contains(&suffix) {
+                    return true;
+                }
+            }
+        }
```

Comment:
Nit: I recommend matching on the extension or file name explicitly. E.g. we don't want to match a file named `py`

```suggestion
	if matches!(path.file_name(), Some("pyproject.toml" | "ruff.toml" |".ruff.toml")) {
		true
	} else if matches!(path.extension(), Some("py" | "pyi")) { 
		true
	} else { false }
```

Reply:
Good catch!